File: fashion_social_app/lib/screens/competitions_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../providers/theme_provider.dart';
import '../providers/competition_provider.dart';
import '../widgets/gradient_text.dart';
import '../themes/app_theme.dart';

class CompetitionsScreen extends StatefulWidget {
  const CompetitionsScreen({super.key});

  @override
  State<CompetitionsScreen> createState() => _CompetitionsScreenState();
}

class _CompetitionsScreenState extends State<CompetitionsScreen>
    with TickerProviderStateMixin {
  late TabController _tabController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(length: 3, vsync: this);
    _loadCompetitions();
  }

  @override
  void dispose() {
    _tabController.dispose();
    super.dispose();
  }

  Future<void> _loadCompetitions() async {
    final competitionProvider = Provider.of<CompetitionProvider>(context, listen: false);
    await competitionProvider.loadCompetitions();
  }

  @override
  Widget build(BuildContext context) {
    final themeProvider = Provider.of<ThemeProvider>(context);
    final competitionProvider = Provider.of<CompetitionProvider>(context);
    final isDark = themeProvider.isDarkMode;

    return Scaffold(
      appBar: AppBar(
        title: GradientText(
          'Competitions',
          gradient: AppTheme.goldGradient,
          style: const TextStyle(
            fontSize: 24,
            fontWeight: FontWeight.bold,
          ),
        ),
        centerTitle: true,
        bottom: TabBar(
          controller: _tabController,
          labelColor: AppTheme.accentColor,
          unselectedLabelColor: Colors.grey,
          indicatorColor: AppTheme.accentColor,
          indicatorWeight: 3,
          tabs: const [
            Tab(
              text: 'Active',
              icon: Icon(Icons.emoji_events),
            ),
            Tab(
              text: 'Upcoming',
              icon: Icon(Icons.upcoming),
            ),
            Tab(
              text: 'Completed',
              icon: Icon(Icons.check_circle),
            ),
          ],
        ),
      ),
      body: TabBarView(
        controller: _tabController,
        children: [
          _buildActiveCompetitions(competitionProvider, isDark),
          _buildUpcomingCompetitions(competitionProvider, isDark),
          _buildCompletedCompetitions(competitionProvider, isDark),
        ],
      ),
    );
  }

  Widget _buildActiveCompetitions(CompetitionProvider competitionProvider, bool isDark) {
    final activeCompetitions = competitionProvider.getActiveCompetitions();

    if (competitionProvider.isLoading && activeCompetitions.isEmpty) {
      return const Center(
        child: CircularProgressIndicator(
          color: AppTheme.accentColor,
        ),
      );
    }

    if (activeCompetitions.isEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(
              Icons.emoji_events_outlined,
              size: 64,
              color: Colors.grey[400],
            ),
            const SizedBox(height: 16),
            Text(
              'No active competitions',
              style: TextStyle(
                fontSize: 18,
                color: Colors.grey[600],
                fontWeight: FontWeight.w500,
              ),
            ),
            const SizedBox(height: 8),
            Text(
              'Check back soon for new challenges!',
              style: TextStyle(
                fontSize: 14,
                color: Colors.grey[500],
              ),
            ),
          ],
        ),
      );
    }

    return RefreshIndicator(
      onRefresh: _loadCompetitions,
      color: AppTheme.accentColor,
      child: ListView.builder(
        padding: const EdgeInsets.all(16),
        itemCount: activeCompetitions.length,
        itemBuilder: (context, index) {
          final competition = activeCompetitions[index];
          return _buildCompetitionCard(competition, isDark, isActive: true);
        },
      ),
    );
  }

  Widget _buildUpcomingCompetitions(CompetitionProvider competitionProvider, bool isDark) {
    final upcomingCompetitions = competitionProvider.getUpcomingCompetitions();

    if (upcomingCompetitions.isEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(
              Icons.upcoming_outlined,
              size: 64,
              color: Colors.grey[400],
            ),
            const SizedBox(height: 16),
            Text(
              'No upcoming competitions',
              style: TextStyle(
                fontSize: 18,
                color: Colors.grey[600],
                fontWeight: FontWeight.w500,
              ),
            ),
            const SizedBox(height: 8),
            Text(
              'New competitions will be announced soon!',
              style: TextStyle(
                fontSize: 14,
                color: Colors.grey[500],
              ),
            ),
          ],
        ),
      );
    }

    return ListView.builder(
      padding: const EdgeInsets.all(16),
      itemCount: upcomingCompetitions.length,
      itemBuilder: (context, index) {
        final competition = upcomingCompetitions[index];
        return _buildCompetitionCard(competition, isDark, isActive: false);
      },
    );
  }

  Widget _buildCompletedCompetitions(CompetitionProvider competitionProvider, bool isDark) {
    final completedCompetitions = competitionProvider.getCompletedCompetitions();

    if (completedCompetitions.isEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(
              Icons.check_circle_outline,
              size: 64,
              color: Colors.grey[400],
            ),
            const SizedBox(height: 16),
            Text(
              'No completed competitions yet',
              style: TextStyle(
                fontSize: 18,
                color: Colors.grey[600],
                fontWeight: FontWeight.w500,
              ),
            ),
            const SizedBox(height: 8),
            Text(
              'Completed competitions will appear here',
              style: TextStyle(
                fontSize: 14,
                color: Colors.grey[500],
              ),
            ),
          ],
        ),
      );
    }

    return ListView.builder(
      padding: const EdgeInsets.all(16),
      itemCount: completedCompetitions.length,
      itemBuilder: (context, index) {
        final competition = completedCompetitions[index];
        return _buildCompetitionCard(competition, isDark, isActive: false);
      },
    );
  }

  Widget _buildCompetitionCard(competition, bool isDark, {required bool isActive}) {
    return Card(
      margin: const EdgeInsets.only(bottom: 16),
      elevation: 12,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(20),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Banner Image
          if (competition.bannerImage != null)
            ClipRRect(
              borderRadius: const BorderRadius.vertical(top: Radius.circular(20)),
              child: Image.network(
                competition.bannerImage!,
                width: double.infinity,
                height: 180,
                fit: BoxFit.cover,
                errorBuilder: (context, error, stackTrace) {
                  return Container(
                    width: double.infinity,
                    height: 180,
                    decoration: BoxDecoration(
                      gradient: AppTheme.primaryGradient,
                      borderRadius: const BorderRadius.vertical(top: Radius.circular(20)),
                    ),
                    child: const Icon(
                      Icons.emoji_events,
                      size: 64,
                      color: Colors.white,
                    ),
                  );
                },
              ),
            ),

          Padding(
            padding: const EdgeInsets.all(20),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Header
                Row(
                  children: [
                    Expanded(
                      child: Text(
                        competition.title,
                        style: TextStyle(
                          fontSize: 20,
                          fontWeight: FontWeight.bold,
                          color: isDark ? Colors.white : AppTheme.primaryColor,
                        ),
                      ),
                    ),
                    Container(
                      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                      decoration: BoxDecoration(
                        color: isActive ? AppTheme.successColor : AppTheme.warningColor,
                        borderRadius: BorderRadius.circular(20),
                      ),
                      child: Text(
                        competition.statusDisplayName,
                        style: const TextStyle(
                          color: Colors.white,
                          fontSize: 12,
                          fontWeight: FontWeight.w600,
                        ),
                      ),
                    ),
                  ],
                ),

                const SizedBox(height: 12),

                // Type
                Container(
                  padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                  decoration: BoxDecoration(
                    color: AppTheme.accentColor.withOpacity(0.1),
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: Text(
                    competition.typeDisplayName,
                    style: TextStyle(
                      color: AppTheme.accentColor,
                      fontSize: 12,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                ),

                const SizedBox(height: 12),

                // Description
                Text(
                  competition.description,
                  style: TextStyle(
                    fontSize: 14,
                    color: isDark ? Colors.white70 : Colors.grey[700],
                    height: 1.4,
                  ),
                  maxLines: 3,
                  overflow: TextOverflow.ellipsis,
                ),

                const SizedBox(height: 16),

                // Sponsor
                if (competition.sponsor != null) ...[
                  Row(
                    children: [
                      Icon(
                        Icons.business,
                        size: 16,
                        color: Colors.grey[600],
                      ),
                      const SizedBox(width: 4),
                      Text(
                        'Sponsored by ${competition.sponsor}',
                        style: TextStyle(
                          fontSize: 12,
                          color: Colors.grey[600],
                          fontWeight: FontWeight.w500,
                        ),
                      ),
                    ],
                  ),
                  const SizedBox(height: 12),
                ],

                // Prize
                if (competition.prizeDescription != null) ...[
                  Container(
                    padding: const EdgeInsets.all(12),
                    decoration: BoxDecoration(
                      gradient: AppTheme.goldGradient,
                      borderRadius: BorderRadius.circular(12),
                    ),
                    child: Row(
                      children: [
                        const Icon(
                          Icons.card_giftcard,
                          color: Colors.white,
                          size: 20,
                        ),
                        const SizedBox(width: 8),
                        Expanded(
                          child: Text(
                            competition.prizeDescription!,
                            style: const TextStyle(
                              color: Colors.white,
                              fontSize: 12,
                              fontWeight: FontWeight.w600,
                            ),
                          ),
                        ),
                      ],
                    ),
                  ),
                  const SizedBox(height: 16),
                ],

                // Stats
                Row(
                  children: [
                    Icon(
                      Icons.people,
                      size: 16,
                      color: Colors.grey[600],
                    ),
                    const SizedBox(width: 4),
                    Text(
                      '${competition.entriesCount} entries',
                      style: TextStyle(
                        fontSize: 12,
                        color: Colors.grey[600],
                        fontWeight: FontWeight.w500,
                      ),
                    ),
                    const Spacer(),
                    Icon(
                      Icons.tag,
                      size: 16,
                      color: Colors.grey[600],
                    ),
                    const SizedBox(width: 4),
                    Text(
                      competition.hashtag,
                      style: TextStyle(
                        fontSize: 12,
                        color: AppTheme.accentColor,
                        fontWeight: FontWeight.w600,
                      ),
                    ),
                  ],
                ),

                const SizedBox(height: 16),

                // Date Range
                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                    color: isDark ? AppTheme.surfaceDark : Colors.grey[100],
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: Row(
                    children: [
                      Icon(
                        Icons.schedule,
                        size: 16,
                        color: Colors.grey[600],
                      ),
                      const SizedBox(width: 8),
                      Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text(
                            'Start: ${_formatDate(competition.startDate)}',
                            style: TextStyle(
                              fontSize: 11,
                              color: Colors.grey[600],
                            ),
                          ),
                          Text(
                            'End: ${_formatDate(competition.endDate)}',
                            style: TextStyle(
                              fontSize: 11,
                              color: Colors.grey[600],
                            ),
                          ),
                        ],
                      ),
                    ],
                  ),
                ),

                const SizedBox(height: 20),

                // Action Button
                SizedBox(
                  width: double.infinity,
                  child: ElevatedButton(
                    onPressed: isActive
                        ? () {
                            // TODO: Enter competition
                          }
                        : null,
                    style: ElevatedButton.styleFrom(
                      backgroundColor: isActive ? AppTheme.accentColor : Colors.grey,
                      foregroundColor: Colors.white,
                      padding: const EdgeInsets.symmetric(vertical: 16),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(12),
                      ),
                    ),
                    child: Text(
                      isActive ? 'Enter Competition' : 'Not Available',
                      style: const TextStyle(
                        fontSize: 16,
                        fontWeight: FontWeight.w600,
                      ),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  String _formatDate(DateTime date) {
    return '${date.day}/${date.month}/${date.year}';
  }
}

File: fashion_social_app/src/screens/CompetitionsScreen.js
import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  FlatList,
  TouchableOpacity,
  StyleSheet,
  Image,
  RefreshControl,
  Dimensions,
  Alert,
} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useCompetition } from '../contexts/CompetitionContext';
import { useTheme } from '../contexts/ThemeContext';

const { width } = Dimensions.get('window');

const CompetitionsScreen = () => {
  const { competitions, isLoading, loadCompetitions, getActiveCompetitions, getUpcomingCompetitions, getCompletedCompetitions } = useCompetition();
  const { theme } = useTheme();
  const [refreshing, setRefreshing] = useState(false);
  const [selectedTab, setSelectedTab] = useState('active');

  useEffect(() => {
    loadCompetitions();
  }, []);

  const onRefresh = async () => {
    setRefreshing(true);
    await loadCompetitions();
    setRefreshing(false);
  };

  const handleEnterCompetition = (competition) => {
    Alert.alert(
      'Enter Competition',
      `Are you sure you want to enter "${competition.title}"?`,
      [
        { text: 'Cancel', style: 'cancel' },
        { text: 'Enter', onPress: () => {
          // TODO: Implement competition entry
          Alert.alert('Coming Soon', 'Competition entry feature will be available soon!');
        }}
      ]
    );
  };

  const renderCompetitionCard = ({ item }) => (
    <View style={[styles.competitionCard, { backgroundColor: theme.colors.card }]}>
      {item.bannerImage && (
        <Image
          source={{ uri: item.bannerImage }}
          style={styles.bannerImage}
        />
      )}
      
      <View style={styles.competitionContent}>
        <View style={styles.competitionHeader}>
          <Text style={[styles.competitionTitle, { color: theme.colors.text }]}>
            {item.title}
          </Text>
          <View style={[
            styles.statusBadge,
            { backgroundColor: getStatusColor(item.status) }
          ]}>
            <Text style={styles.statusText}>
              {getStatusLabel(item.status)}
            </Text>
          </View>
        </View>

        <View style={styles.typeContainer}>
          <Text style={styles.typeText}>
            {getTypeLabel(item.type)}
          </Text>
        </View>

        <Text style={[styles.description, { color: theme.colors.textSecondary }]}>
          {item.description}
        </Text>

        {item.sponsor && (
          <View style={styles.sponsorContainer}>
            <Text style={styles.sponsorIcon}>🏢</Text>
            <Text style={styles.sponsorText}>
              Sponsored by {item.sponsor}
            </Text>
          </View>
        )}

        {item.prizeDescription && (
          <LinearGradient
            colors={['#FFD700', '#FFA000']}
            style={styles.prizeContainer}
          >
            <Text style={styles.prizeIcon}>🏆</Text>
            <Text style={styles.prizeText}>
              {item.prizeDescription}
            </Text>
          </LinearGradient>
        )}

        <View style={styles.statsContainer}>
          <View style={styles.statItem}>
            <Text style={styles.statIcon}>👥</Text>
            <Text style={[styles.statText, { color: theme.colors.textSecondary }]}>
              {item.entriesCount} entries
            </Text>
          </View>
          <View style={styles.statItem}>
            <Text style={styles.statIcon}>🏷️</Text>
            <Text style={styles.hashtagText}>
              {item.hashtag}
            </Text>
          </View>
        </View>

        <View style={styles.dateContainer}>
          <Text style={styles.dateIcon}>📅</Text>
          <View style={styles.dateInfo}>
            <Text style={[styles.dateText, { color: theme.colors.textSecondary }]}>
              Start: {formatDate(item.startDate)}
            </Text>
            <Text style={[styles.dateText, { color: theme.colors.textSecondary }]}>
              End: {formatDate(item.endDate)}
            </Text>
          </View>
        </View>

        <TouchableOpacity
          style={[
            styles.enterButton,
            { 
              backgroundColor: item.status === 'active' ? '#E91E63' : theme.colors.border,
              opacity: item.status === 'active' ? 1 : 0.6
            }
          ]}
          onPress={() => item.status === 'active' && handleEnterCompetition(item)}
          disabled={item.status !== 'active'}
        >
          <Text style={[
            styles.enterButtonText,
            { color: item.status === 'active' ? '#fff' : theme.colors.textSecondary }
          ]}>
            {getButtonText(item.status)}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );

  const getStatusColor = (status) => {
    switch (status) {
      case 'active': return '#4CAF50';
      case 'upcoming': return '#FF9800';
      case 'judging': return '#2196F3';
      case 'completed': return '#9E9E9E';
      default: return '#9E9E9E';
    }
  };

  const getStatusLabel = (status) => {
    switch (status) {
      case 'active': return 'ACTIVE';
      case 'upcoming': return 'UPCOMING';
      case 'judging': return 'JUDGING';
      case 'completed': return 'COMPLETED';
      default: return 'UNKNOWN';
    }
  };

  const getTypeLabel = (type) => {
    switch (type) {
      case 'dailyTheme': return 'Daily Theme';
      case 'brandCollab': return 'Brand Collaboration';
      case 'weeklyChallenge': return 'Weekly Challenge';
      default: return 'Competition';
    }
  };

  const getButtonText = (status) => {
    switch (status) {
      case 'active': return 'Enter Competition';
      case 'upcoming': return 'Notify Me';
      case 'judging': return 'Judging in Progress';
      case 'completed': return 'View Results';
      default: return 'Not Available';
    }
  };

  const formatDate = (dateString) => {
    const date = new Date(dateString);
    return `${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`;
  };

  const getCompetitionsForTab = () => {
    switch (selectedTab) {
      case 'active':
        return getActiveCompetitions();
      case 'upcoming':
        return getUpcomingCompetitions();
      case 'completed':
        return getCompletedCompetitions();
      default:
        return [];
    }
  };

  return (
    <View style={[styles.container, { backgroundColor: theme.colors.background }]}>
      <View style={styles.header}>
        <Text style={[styles.headerTitle, { color: theme.colors.text }]}>
          Competitions
        </Text>
      </View>

      <View style={styles.tabContainer}>
        <TouchableOpacity
          style={[
            styles.tab,
            selectedTab === 'active' && styles.activeTab,
            { borderColor: theme.colors.border }
          ]}
          onPress={() => setSelectedTab('active')}
        >
          <Text style={[
            styles.tabText,
            selectedTab === 'active' && styles.activeTabText,
            { color: selectedTab === 'active' ? theme.colors.accent : theme.colors.textSecondary }
          ]}>
            Active
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[
            styles.tab,
            selectedTab === 'upcoming' && styles.activeTab,
            { borderColor: theme.colors.border }
          ]}
          onPress={() => setSelectedTab('upcoming')}
        >
          <Text style={[
            styles.tabText,
            selectedTab === 'upcoming' && styles.activeTabText,
            { color: selectedTab === 'upcoming' ? theme.colors.accent : theme.colors.textSecondary }
          ]}>
            Upcoming
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[
            styles.tab,
            selectedTab === 'completed' && styles.activeTab,
            { borderColor: theme.colors.border }
          ]}
          onPress={() => setSelectedTab('completed')}
        >
          <Text style={[
            styles.tabText,
            selectedTab === 'completed' && styles.activeTabText,
            { color: selectedTab === 'completed' ? theme.colors.accent : theme.colors.textSecondary }
          ]}>
            Completed
          </Text>
        </TouchableOpacity>
      </View>

      <FlatList
        data={getCompetitionsForTab()}
        renderItem={renderCompetitionCard}
        keyExtractor={(item) => item.id}
        refreshControl={
          <RefreshControl
            refreshing={refreshing}
            onRefresh={onRefresh}
            colors={['#E91E63']}
          />
        }
        contentContainerStyle={styles.competitionsContainer}
        ListEmptyComponent={
          <View style={styles.emptyState}>
            <Text style={styles.emptyIcon}>
              {selectedTab === 'active' ? '🏆' : selectedTab === 'upcoming' ? '📅' : '✅'}
            </Text>
            <Text style={[styles.emptyTitle, { color: theme.colors.text }]}>
              No {selectedTab} competitions
            </Text>
            <Text style={styles.emptyDescription}>
              {selectedTab === 'active' 
                ? 'Check back soon for new challenges!'
                : selectedTab === 'upcoming'
                ? 'New competitions will be announced soon!'
                : 'Completed competitions will appear here'
              }
            </Text>
          </View>
        }
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(0,0,0,0.1)',
  },
  headerTitle: {
    fontSize: 24,
    fontWeight: 'bold',
  },
  tabContainer: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(0,0,0,0.1)',
  },
  tab: {
    flex: 1,
    paddingVertical: 16,
    alignItems: 'center',
    borderBottomWidth: 2,
  },
  activeTab: {
    borderBottomColor: '#E91E63',
  },
  tabText: {
    fontSize: 14,
    fontWeight: '500',
  },
  activeTabText: {
    color: '#E91E63',
  },
  competitionsContainer: {
    padding: 16,
    gap: 16,
  },
  competitionCard: {
    borderRadius: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.1,
    shadowRadius: 12,
    elevation: 8,
    overflow: 'hidden',
  },
  bannerImage: {
    width: '100%',
    height: 180,
  },
  competitionContent: {
    padding: 20,
  },
  competitionHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginBottom: 12,
  },
  competitionTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    flex: 1,
    marginRight: 12,
  },
  statusBadge: {
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 20,
  },
  statusText: {
    color: '#fff',
    fontSize: 12,
    fontWeight: '600',
  },
  typeContainer: {
    alignSelf: 'flex-start',
    marginBottom: 12,
  },
  typeText: {
    backgroundColor: 'rgba(233, 30, 99, 0.1)',
    color: '#E91E63',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 12,
    fontSize: 12,
    fontWeight: '600',
  },
  description: {
    fontSize: 14,
    lineHeight: 20,
    marginBottom: 16,
  },
  sponsorContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 16,
  },
  sponsorIcon: {
    fontSize: 16,
    marginRight: 8,
  },
  sponsorText: {
    fontSize: 12,
    color: '#666',
    fontWeight: '500',
  },
  prizeContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 12,
    borderRadius: 12,
    marginBottom: 16,
  },
  prizeIcon: {
    fontSize: 20,
    marginRight: 12,
  },
  prizeText: {
    color: '#fff',
    fontSize: 12,
    fontWeight: '600',
    flex: 1,
  },
  statsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 16,
  },
  statItem: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  statIcon: {
    fontSize: 16,
    marginRight: 4,
  },
  statText: {
    fontSize: 12,
    fontWeight: '500',
  },
  hashtagText: {
    fontSize: 12,
    color: '#E91E63',
    fontWeight: '600',
  },
  dateContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.05)',
    padding: 12,
    borderRadius: 12,
    marginBottom: 20,
  },
  dateIcon: {
    fontSize: 16,
    marginRight: 8,
  },
  dateInfo: {
    flex: 1,
  },
  dateText: {
    fontSize: 11,
  },
  enterButton: {
    paddingVertical: 16,
    borderRadius: 12,
    alignItems: 'center',
  },
  enterButtonText: {
    fontSize: 16,
    fontWeight: '600',
  },
  emptyState: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 60,
  },
  emptyIcon: {
    fontSize: 64,
    marginBottom: 16,
  },
  emptyTitle: {
    fontSize: 18,
    fontWeight: '600',
    marginBottom: 8,
  },
  emptyDescription: {
    fontSize: 14,
    color: '#666',
    textAlign: 'center',
  },
});

export default CompetitionsScreen;